CSV sample fixes

datalogics-coreyy opened the pull request:
#### Changes in this pull request

- Excludes the ExportFormDataToCsv sample from Talkeetna 5
- Refactors method for checking PDFJT version into VersionUtils
- Shrinks the Talkeetna 6 profile range from starting at 6.0.0 to 6.0.11

#### Checklist for approving this pull request

(PR creator amend this with more conditions if necessary)

- [x] There are **unit tests** for new/changed code, or a good explanation why this was not possible.
- [x] All **CI builders** have indicated success (Give them a few minutes to notice the pull request.)
- [x] The **Pull request title** has the JIRA issue numbers separated by spaces (if any), a space, and then a short, but descriptive summary.
- [x] **Commit messages** are well formed: [A note about Git commit messages](http://www.tpope.net/node/106)
- [x] New public packages, classes, and methods are **documented**. (Strongly consider documenting private classes and methods.)

#### Blockers

Add a checkbox for yourself with your **&#64;name** to this list if you are holding this PR open for review. PR Shepherd, please hold off merging this PR until these are all checked:

- [x] _&lt;add your name here with an **@**>_

datalogics-bhaugen committed:
Add VersionUtils class

- move the pdfjtIsBeforeVersion4 method here

datalogics-devauto commented:
Build succeeded.

datalogics-devauto commented:
Build succeeded.

datalogics-bhaugen commented on src/main/java/com/datalogics/pdf/samples/forms/ExportFormDataToCsv.java:
```diff
@@ -155,26 +156,48 @@ private static void exportFieldValues(final PDFInteractiveForm form, final CSVPr
+     * @throws PDFSecurityException some general security issue occurred during the processing of the request
+     * @throws IOException an I/O operation failed or was interrupted
+     */
+    private static void validateField(final CSVPrinter printer, final PDFField field, final List value)
```
This method does more than just check for a null value

datalogics-bhaugen commented on src/main/java/com/datalogics/pdf/samples/forms/ExportFormDataToCsv.java:
```diff
@@ -155,26 +156,48 @@ private static void exportFieldValues(final PDFInteractiveForm form, final CSVPr
+                value = field.getValueList();
+                validateField(printer, field, value);
+            } else {
+                final String formattedValue = field.getFormattedValue();
```
Since the _getFormattedValue()_ API did not exist in PDF Java Toolkit 4.0.0 (it didn't show up until 4.7.0), it would really be nice to check the version number more accurately in this case.

datalogics-coreyy replied:
@datalogics-bhaugen I could create a method in VersionUtils that checks specifically for version 4.7.0 . Is that what you were thinking?

datalogics-bhaugen replied:
@datalogics-coreyy, I didn't have anything specific in mind. Thinking about it a little more though and having a method for each version could end up with a large number of methods. Maybe having a method where the sample can pass a version string to test would be a different solution to consider? That way there is one method additional method and samples could modify their behavior granularly.

datalogics-devauto commented:
Build succeeded.

datalogics-devauto commented:
Build succeeded.

datalogics-coreyy committed:
Add utility method to check for the current version of PDFJT

- If the version string passed in matches the version string of the
  current PDFJT, return true. Otherwise, return false.

datalogics-devauto commented:
Build succeeded.

datalogics-devauto commented:
Build succeeded.

datalogics-devauto commented:
Build succeeded.

datalogics-devauto commented:
Build succeeded.

datalogics-devauto commented:
Build failed.

datalogics-devauto commented:
Build succeeded.

datalogics-devauto commented:
Build succeeded.

datalogics-devauto commented:
Build succeeded.

datalogics-devauto commented:
Build succeeded.

datalogics-devauto commented:
Build succeeded.

datalogics-coreyy committed:
Shorten Talkeetna version range for use-talkeetna-6 profile

- We’re adding a sample that uses a PDFJT API that wasn’t present until
  Talkeetna 6.0.11 (PDFJT 4.7.1)

datalogics-coreyy committed:
Exclude ExportFormDataToCsv sample for Talkeetna 5

- This will exclude the sample, its test, and its documentation

datalogics-devauto commented:
Build failed.

datalogics-devauto commented:
Build failed.

datalogics-coreyy committed:
Provide private constructor to avoid PMD error

datalogics-devauto commented:
Build failed.

datalogics-devauto commented:
Build succeeded.

datalogics-kam commented on pom.xml:
```diff
@@ -164,9 +164,11 @@
           <target>1.7</target>
           <excludes>
             <exclude>com/datalogics/pdf/samples/extraction/TextCharacterBoxes.java</exclude>
+	    <exclude>com/datalogics/pdf/samples/forms/ExportFormDataToCsv.java</exclude>
```
This fails on CI because the ``pom.xml`` changes during the build.

- Don't use tabs in XML, use spaces.
- Run a build or ``sortpom:sort`` before checking in POM changes. Even just building once with ``mvn`` will put all the POMs into canonical form.
- Please correct this and rebase the fix into this commit.

datalogics-kam replied:
This is the part of the Jenkins log that tells you that the POM was changed during building. It's near the end:

```
[workspace] $ /bin/sh -xe /tmp/hudson4715329650434021097.sh
+ git diff --stat --exit-code
 pom.xml | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)
+ echo '[ERROR] Build changed some Git files'
[ERROR] Build changed some Git files
+ exit 1
Build step 'Execute shell' marked build as failure
```

datalogics-coreyy replied:
Thanks! I'll definitely keep the sortpom in mind in the future.

datalogics-kam commented on pom.xml:
```diff
@@ -659,12 +659,12 @@
             <groupId>com.datalogics.pdf</groupId>
             <artifactId>talkeetna</artifactId>
-            <version>[6.0.0-SNAPSHOT,7.0.0-SNAPSHOT)</version>
+            <version>[6.0.11,7.0.0-SNAPSHOT)</version>
```
These changes should also be made into ``lite/pom.xml``

datalogics-kam commented on src/main/java/com/datalogics/pdf/samples/util/VersionUtils.java:
```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright 2015 Datalogics, Inc.
```
Please change this to the inception date of 2017.

datalogics-kam commented on src/main/java/com/datalogics/pdf/samples/util/VersionUtils.java:
```diff
@@ -0,0 +1,63 @@
+     * @throws IOException an I/O operation failed or was interrupted
+     */
+    public static boolean pdfjtIsBeforeVersion4() throws IOException {
+        try (final InputStream propertiesStream = Version.class.getResourceAsStream("version.properties")) {
```
This is practically a duplicate of ``pdfjtIsBeforeVersion``. Please just call that function with ``"4.0.0-SNAPSHOT"`` and eliminate the duplicate code.

datalogics-kam commented on src/main/java/com/datalogics/pdf/samples/util/VersionUtils.java:
```diff
@@ -0,0 +1,63 @@
+     * Checks if the version string passed in matches the string for the current PDFJT.
+     *
+     * @param versionToCompare String specifying version to check against
+     * @return is this the current version of PDFJT
```
How about "...true if the version of PDFJT is before the version specified as an argument" ?

datalogics-coreyy commented:
trigger build

datalogics-kam commented on src/test/java/com/datalogics/pdf/samples/forms/FillFormTest.java:
```diff
@@ -122,7 +123,7 @@ public void testAcroformXfdf() throws Exception {
         assertTrue(outputPdfFile.getPath() + " must exist after run", outputPdfFile.exists());
         checkForms(outputPdfFile.toURI().toURL(),
-                   FormImporter.pdfjtIsBeforeVersion4() ? ACROFORM_XFDF_DATA_PDFJT_3 : ACROFORM_XFDF_DATA);
+                   VersionUtils.pdfjtIsBeforeVersion4() ? ACROFORM_XFDF_DATA_PDFJT_3 : ACROFORM_XFDF_DATA);
```
I'm really glad you refactored this to uncouple the samples. Thanks.

datalogics-devauto commented:
Build succeeded.

datalogics-devauto commented:
Build failed.

datalogics-coreyy committed:
Remove duplicate code

- With the addition of the method that allows checking for specific
  Version of PDFJT, we no longer need pdfjtIsBefore4().

datalogics-devauto commented:
Build succeeded.

datalogics-devauto commented:
Build succeeded.

datalogics-kam commented:
Awesome. Thanks for fixing this up! 👍 🐟
